dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/WeaponPart.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package dk.sdu.se4.common.entity.part;
+
+import dk.sdu.se4.common.entity.Entity;
+
+/**
+ *
+ * @author kjalris and denbedste
+ */
+public class WeaponPart implements EntityPart {
+    private boolean type;
+    private int totalBullets;
+    private int magazineSize;
+    private int currentMagazine;
+    private int fireRate;
+
+    public WeaponPart(boolean type, int totalBullets, int magazineSize, int fireRate) {
+        this.type = type;
+        this.totalBullets = totalBullets;
+        this.magazineSize = magazineSize;
+        this.currentMagazine = magazineSize;
+        this.fireRate = fireRate;
+    }
+
+    public boolean getType() {
+        return type;
+    }
+    
+    public void setType(boolean type) {
+        this.type = type;
+    }
+    
+    public void removeBullet() {
+        totalBullets -= 1;
+        currentMagazine -= 1;
+    }
+    
+    public void addBullets(int bullets) {
+        this.totalBullets = bullets;
+    }
+
+    public int getCurrentMagazine() {
+        return currentMagazine;
+    }
+
+    public int getMagazineSize() {
+        return magazineSize;
+    }
+
+    public int getTotalBullets() {
+        return totalBullets;
+    }
+
+    public void reload() {
+        currentMagazine = magazineSize;
```

Comment:
"totalBullets" is never used. I guess it should be used here, only allowing you to reload all the bullets you have and up to max mag size.

Reply:
TotalBullets is also used for (if totalbullets > 0) to be able to only shoot if you have bullets left (Entity bullet = new Bullet())
Could make a method: public boolean isBulletsLeft() which return if there are more than 0 totalBullets, but a method to exchange if(wp.getTotalBullets == 0) to if(wp.isBulletsLeft()) shouldnt be neccesary